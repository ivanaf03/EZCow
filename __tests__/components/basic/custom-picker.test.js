import React from "react";

import { render } from "@testing-library/react-native";

import CustomPicker from "../../../components/basic/custom-picker";

describe("CustomPicker", () => {

    const mockOnValueChange = jest.fn();
    const mockOptions = [
        { label: 'Option 1', value: 'option1' },
        { label: 'Option 2', value: 'option2' },
        { label: 'Option 3', value: 'option3' },
        { label: 'Option 4', value: 'option4' },
        { label: 'Option 5', value: 'option5' },
    ];

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", () => {
        const tree = render(
            <CustomPicker
                text={'test'}
                value={'test'}
                onValueChange={mockOnValueChange}
                options={mockOptions}
            />
        );
        expect(tree).toMatchSnapshot();
    });

});