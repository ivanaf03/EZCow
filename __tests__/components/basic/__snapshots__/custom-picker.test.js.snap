// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPicker should render correctly 1`] = `
<View
  style={
    [
      {
        "marginTop": -4,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "fontFamily": "Nunito-Light",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    test
    :
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#ffffff",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#2254ba",
          "borderLeftColor": "#2254ba",
          "borderRightColor": "#2254ba",
          "borderTopColor": "#2254ba",
        },
        {
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
        },
        {
          "fontFamily": "Nunito-Medium",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    <View>
      <RNCPicker
        items={
          [
            {
              "label": {
                "label": "Option 1",
                "value": "option1",
              },
              "testID": undefined,
              "textColor": undefined,
              "value": {
                "label": "Option 1",
                "value": "option1",
              },
            },
            {
              "label": {
                "label": "Option 2",
                "value": "option2",
              },
              "testID": undefined,
              "textColor": undefined,
              "value": {
                "label": "Option 2",
                "value": "option2",
              },
            },
            {
              "label": {
                "label": "Option 3",
                "value": "option3",
              },
              "testID": undefined,
              "textColor": undefined,
              "value": {
                "label": "Option 3",
                "value": "option3",
              },
            },
            {
              "label": {
                "label": "Option 4",
                "value": "option4",
              },
              "testID": undefined,
              "textColor": undefined,
              "value": {
                "label": "Option 4",
                "value": "option4",
              },
            },
            {
              "label": {
                "label": "Option 5",
                "value": "option5",
              },
              "testID": undefined,
              "textColor": undefined,
              "value": {
                "label": "Option 5",
                "value": "option5",
              },
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          [
            {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
